lib/ngx/balancer.lua

```diff
@@ -1,10 +1,11 @@
 -- Copyright (C) Yichun Zhang (agentzh)
 
 
-local ffi = require "ffi"
 local base = require "resty.core.base"
+base.check_subsystem('http', 'stream')
```

Comment:
The `check_subsystem` is a little bit confusing. Maybe `allows_subsystem()` is a better name? Better be more explicit about our intention (like allows or not allows).